src/jit/gentree.cpp

```diff
@@ -5990,37 +5990,71 @@ GenTreeBlkOp* Compiler::gtNewCpObjNode(GenTreePtr dst,
 
     GenTreeBlkOp* result = nullptr;
     
-    // Get the GC fields info
-    size = info.compCompHnd->getClassSize(structHnd);
+    bool useCopyObj = false;
 
-    bool      useCopyObj = false;
+    if (structHnd == nullptr)
+    {
+#if FEATURE_SIMD
+        assert(src->OperGet() == GT_ADDR);
+
+        GenTree* srcValue = src->gtGetOp1();
+
+        type = srcValue->TypeGet();
+        assert(varTypeIsSIMD(type));
+
+        switch (type)
```

Comment:
Please add a comment line describing the reason for this logic (that was mentioned in PR description)